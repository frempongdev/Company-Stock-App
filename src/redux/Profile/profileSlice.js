import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';

const profileBaseUrl = 'https://financialmodelingprep.com/api/v3/profile/';
const apikey = '?apikey=958d9faee6392a89d333869c614c50e6';

const initialState = {
  profile: [],
  isLoading: false,
};

export const fetchProfile = createAsyncThunk('countries/fetchProfile', async (companySymbol) => {
  const response = await fetch(`${profileBaseUrl}${companySymbol}${apikey}`);
  const data = response.json();
  return data;
});

const profileSlice = createSlice({
  name: 'profile',
  initialState,
  reducers: {},
  extraReducers: (builder) => {
    builder
      .addCase(fetchProfile.pending, (state) => ({ ...state, isLoading: true }));
  },
});

export default profileSlice.reducer;
