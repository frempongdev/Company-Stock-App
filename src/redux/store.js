import { configureStore } from '@reduxjs/toolkit';
import companiesReducer from './Companies/CompaniesSlice';

const store = configureStore({
  reducer: {
    companies: companiesReducer,
  },
});

export default store;
