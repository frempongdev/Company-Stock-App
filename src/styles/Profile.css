.profile-main {
    padding: 0% 4%;
}
.back {
    margin-top: 1%;
    float: left;
}
.company-img {
    margin-top: 2%;
    border: 1px solid white;

}
.details-title {
    padding: 2%;
    border: solid white;
}
.details {
    border: solid white;
    padding: 1%;
}