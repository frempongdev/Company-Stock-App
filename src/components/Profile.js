import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams, useNavigate } from 'react-router-dom';
import { fetchProfile } from '../redux/Profile/profileSlice';
import '../styles/Profile.css';

const Profile = () => {
  const { profile, isLoading } = useSelector((state) => state.profile);
  const dispatch = useDispatch();
  const { companySymbol } = useParams();
  const history = useNavigate();

  useEffect(() => {
    dispatch(fetchProfile(companySymbol));
  }, [dispatch]); // eslint-disable-line react-hooks/exhaustive-deps

  if (isLoading) {
    return <div>Loading Company details...</div>;
  }

  return (
    <div className="profile-main">
      <button type="button" className="back" onClick={() => history(-1)}>
        {'<'}
        back
      </button>
      <br />
      <div className="company-img">
        <img src={profile[0]?.image} alt="companyImg" />
      </div>
      <div className="company-details">
        <h2 className="company-name">
          {profile[0]?.companyName}
        </h2>
        <h6 className="company-location">
          {profile[0]?.city}
          ,
          {' '}
          {profile[0]?.country}
        </h6>
        <p className="company-symbol">
          Stock Symbol:
          {' '}
          {profile[0]?.symbol}
        </p>
        <h4 className="stock">
          Stock Price:
          {' '}
          $
          {profile[0]?.price}
        </h4>
        <p className="industry">
          Industry :
          {' '}
          {profile[0]?.industry}
        </p>
        <h4 className="details-title">Company Overview</h4>
        <p className="details">{profile[0]?.description}</p>

      </div>

    </div>
  );
};

export default Profile;
