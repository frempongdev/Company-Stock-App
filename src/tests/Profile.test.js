import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import thunk from 'redux-thunk';
import { render } from '@testing-library/react';
// import { configureMockStore } from '@reduxjs/toolkit';
import configureStore from 'redux-mock-store';

import Profile from '../components/Profile';
import store from '../redux/store';

describe('Profile', () => {
  it('renders Profile', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Profile />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('displays "Loading Company details..." when isLoading is true', () => {
    const mockStore = configureStore([thunk]);

    const store = mockStore({
      profile: {
        profile: [],
        isLoading: true,
      },
    });
    const { getByText } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Profile />
        </BrowserRouter>
      </Provider>,
    );
    const loadingMessage = getByText(/Loading Company details/i);
    expect(loadingMessage).toBeInTheDocument();
  });
});
