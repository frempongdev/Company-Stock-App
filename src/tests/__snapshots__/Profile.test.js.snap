// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile renders Profile 1`] = `
<div>
  <button
    className="back"
    onClick={[Function]}
    type="button"
  >
    &lt;
    back
  </button>
  <div
    className="company-img"
  >
    <img
      alt="companyImg"
    />
  </div>
  <div
    className="company-details"
  >
    <h2
      className="company-name"
    />
    <h6
      className="company-location"
    >
      ,
       
    </h6>
    <p
      className="company-symbol"
    >
      Stock Symbol:
       
    </p>
    <h4
      className="stock"
    >
      Stock Price:
       
      $
    </h4>
    <p
      className="industry"
    >
      Industry :
       
    </p>
    <h4
      className="details"
    >
      Company Overview
    </h4>
    <p
      className="details"
    />
  </div>
</div>
`;
